Apparent array bounds checking warning-as-error false positive with GCC 9.2

Building JerryScript Zephyr port with the latest Zephyr SDK 0.11.1, based on GCC 9.2.0, I get an error like:

~~~
In file included from /mnt/hdd/opt/zephyr-sdk-0.11.1/x86_64-zephyr-elf/x86_64-zephyr-elf/include/string.h:180,
                 from /home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/jrt/jrt.h:20,
                 from /home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/ecma/base/ecma-globals.h:20,
                 from /home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/debugger/debugger.h:19,
                 from /home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/api/jerry.c:18:
/home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/api/jerry.c: In function 'jerry_init':
/home/pfalcon/projects-3rdparty/Embedded/Zephyr/jerryscript/jerry-core/api/jerry.c:194:3: error: '__builtin_memset' offset [17, 2144] from the object at 'jerry_global_context' is out of the bounds of referenced subobject 'ecma_builtin_objects' with type 'jmem_cpointer_t[8]' {aka 'short unsigned int[8]'} at offset 0 [-Werror=array-bounds]
  194 |   memset ((void*)(long) &JERRY_CONTEXT (JERRY_CONTEXT_FIRST_MEMBER), 0,
      |   ^~~~~~
~~~

This is an obvious false positive, with a compiler trying to outsmart a programmer in its understanding of the program being compiled, and failing. But given the pervasive love for `-Wall`, this is another problem to work around.

Full build log can be seen at https://ci.linaro.org/view/lite-iot-ci/job/lite-aeolus-jerryscript/973/PLATFORM=disco_l475_iot1,ZEPHYR_GCC_VARIANT=zephyr,label=docker-xenial-amd64-13/console .

###### JerryScript revision
This happens with the latest master, 0d7f26e2c4e7eb57ed277d112dc5bea663111a64.

###### Build platform
Zephyr port, with Zephyr 4f1626851b8296a86edb67442d56f4ed5f17dc47 and Zephyr SDK 0.11.1 with:

~~~
$ /home/pfalcon/opt/zephyr-sdk-0.11.1/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc --version
x86_64-zephyr-elf-gcc (crosstool-NG 1.24.0.37-3f461da-dirty) 9.2.0
~~~

###### Build steps
Per the Zephyr port instructions:

~~~
make -f ./targets/zephyr/Makefile.zephyr BOARD=qemu_x86
~~~


###### Build log
See above.

--- a/jerry-core/api/jerry.c
+++ b/jerry-core/api/jerry.c
@@ -191,7 +191,7 @@ jerry_init (jerry_init_flag_t flags) /**< combination of Jerry flags */
   JERRY_ASSERT (!(JERRY_CONTEXT (status_flags) & ECMA_STATUS_API_AVAILABLE));
 
   /* Zero out all non-external members. */
-  memset (&JERRY_CONTEXT (JERRY_CONTEXT_FIRST_MEMBER), 0,
+  memset ((char *) &JERRY_CONTEXT_STRUCT + offsetof (jerry_context_t, JERRY_CONTEXT_FIRST_MEMBER), 0,
           sizeof (jerry_context_t) - offsetof (jerry_context_t, JERRY_CONTEXT_FIRST_MEMBER));
 
   JERRY_CONTEXT (jerry_init_flags) = flags;

--- a/jerry-core/jcontext/jcontext.h
+++ b/jerry-core/jcontext/jcontext.h
@@ -245,6 +245,7 @@ struct jerry_context_t
  * This part is for JerryScript which uses external context.
  */
 
+#define JERRY_CONTEXT_STRUCT (*jerry_port_get_current_context ())
 #define JERRY_CONTEXT(field) (jerry_port_get_current_context ()->field)
 
 #if !ENABLED (JERRY_SYSTEM_ALLOCATOR)
@@ -273,6 +274,11 @@ struct jmem_heap_t
  * Global context.
  */
 extern jerry_context_t jerry_global_context;
+
+/**
+ * Config-independent name for context.
+ */
+#define JERRY_CONTEXT_STRUCT (jerry_global_context)
 
 /**
  * Provides a reference to a field in the current context.

